OracleHTTPServer/dockerfiles/12.2.1.3.0/container-scripts/provisionOHS.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+# Author: prabhat.kishore@oracle.com
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
+#
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+#
+#*************************************************************************
+#  This script is used to create a standalone OHS domain and start NodeManager, OHS instance.
+#  This script sets the following variables:
+#
+#  WL_HOME    - The Weblogic home directory
+#  NODEMGR_HOME  - Absolute path to Nodemanager directory under the configured domain home
+#  DOMAIN_HOME - Absolute path to configured domain home
+#  JAVA_HOME- Absolute path to jre inside the oracle home directory
+#*************************************************************************
+########### SIGTERM handler ############
+function _term() {
+   echo "Stopping container."
+   echo "SIGTERM received, shutting down the server!"
+   ${WLST_HOME}/wlst.sh /u01/oracle/container-scripts/stop-ohs.py
+   ${DOMAIN_HOME}/bin/stopNodeManager.sh
+}
+
+########### SIGKILL handler ############
+function _kill() {
+   echo "SIGKILL received, shutting down the server!"
+   kill -9 $childPID
+}
+
+# Set SIGTERM handler
+trap _term SIGTERM
+
+# Set SIGKILL handler
+trap _kill SIGKILL
+
+echo "MW_HOME=${MW_HOME:?"Please set MW_HOME"}"
+echo "ORACLE_HOME=${ORACLE_HOME:?"Please set ORACLE_HOME"}"
+echo "DOMAIN_NAME=${DOMAIN_NAME:?"Please set DOMAIN_NAME"}"
+echo "OHS_COMPONENT_NAME=${OHS_COMPONENT_NAME:?"Please set OHS_COMPONENT_NAME"}"
+
+export MW_HOME ORACLE_HOME DOMAIN_NAME OHS_COMPONENT_NAME
+
+
+#Set WL_HOME, WLST_HOME, DOMAIN_HOME and NODEMGR_HOME
+WL_HOME=${ORACLE_HOME}/wlserver
+WLST_HOME=${ORACLE_HOME}/oracle_common/common/bin
+echo "WLST_HOME=${WLST_HOME}"
+
+DOMAIN_HOME=${ORACLE_HOME}/user_projects/domains/${DOMAIN_NAME}
+export DOMAIN_HOME
+echo "DOMAIN_HOME=${DOMAIN_HOME}"
+
+NODEMGR_HOME=${DOMAIN_HOME}/nodemanager
+export NODEMGR_HOME
+
+echo "PATH=${PATH}"
+PATH=$PATH:/usr/java/default/bin:/u01/oracle/ohssa/oracle_common/common/bin
+export PATH
+echo "PATH=${PATH}"
+
+#  Set JAVA_OPTIONS and JAVA_HOME for node manager
+JAVA_OPTIONS="${JAVA_OPTIONS} -Dweblogic.RootDirectory=${DOMAIN_HOME}"
+export JAVA_OPTIONS
+
+JAVA_HOME=${ORACLE_HOME}/oracle_common/jdk/jre
+export JAVA_HOME
+
+#Declare and initializing NMSTATUS
+declare -a NMSTATUS
+NMSTATUS[0]="NOT RUNNING"
+
+# If nodemanager$$.log does not exists,this is the first time configuring the NM 
+# generate the NM password 
+
+if [ !  -f /u01/oracle/logs/nodemanager$$.log ]; then
+    
+# Auto generate Node Manager  password
```

Comment:
For security reasons it  is no longer allowed to auto generate password since anyone logging into image or looking at logs can see the password in clear text.
You need to follow the example of the WebLogic 12.2.1.3 image what the password is passed in through a properties files.  The security of the properties file is up to the user but you must put a comment in the readme.  Look at PR https://github.com/oracle/docker-images/pull/1048/files